From 010e1f3a22d98186422d61ea8e818881451e90db Mon Sep 17 00:00:00 2001
From: Sergio Prado <sergio.prado@toradex.com>
Date: Mon, 29 May 2023 08:15:56 -0300
Subject: [PATCH] bootm: don't use HAB to authenticate FIT image

If U-Boot FIT image support is compiled in, do not use HAB to
authenticate image; instead assume FIT signature validation will be
used for image authentication, when needed.

Upstream-Status: Inappropriate [TorizonCore specific]

Signed-off-by: Sergio Prado <sergio.prado@toradex.com>
Signed-off-by: Rogerio Guerra Borin <rogerio.borin@toradex.com>
---
 cmd/bootm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmd/bootm.c b/cmd/bootm.c
index 0c5585b123a..34152861a9b 100644
--- a/cmd/bootm.c
+++ b/cmd/bootm.c
@@ -125,7 +125,7 @@ int do_bootm(struct cmd_tbl *cmdtp, int flag, int argc, char *const argv[])
 			return do_bootm_subcommand(cmdtp, flag, argc, argv);
 	}
 
-#ifdef CONFIG_IMX_HAB
+#if defined(CONFIG_IMX_HAB) && !defined(CONFIG_FIT)
 	extern int authenticate_image(
 			uint32_t ddr_start, uint32_t raw_image_size);
 
-- 
2.25.1

